From bffa5e898cca5a8501dc1fcb75d6331787f9c6e6 Mon Sep 17 00:00:00 2001
From: Balvinder Singh <bpsingh@codeaurora.org>
Date: Mon, 19 Jan 2015 15:41:36 +0530
Subject: [PATCH 24/25] Bluetooth: Avoid Spurious AMP node access if not
 created

Added checks for avoiding the AMP nodes, incase of Spurious
access like data corruption during Rx from BT SOC

Change-Id: Ifc974750efe2082420f89f32580ed8ccd1924a3d
Signed-off-by: Balvinder Singh <bpsingh@codeaurora.org>
CRs-Fixed: 783934
---
 net/bluetooth/hci_event.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index 2d55119..d1dddbf1 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -3408,6 +3408,7 @@ static void hci_phy_link_complete_evt(struct hci_dev *hdev,
 {
 	struct hci_ev_phy_link_complete *ev = (void *) skb->data;
 	struct hci_conn *hcon, *bredr_hcon;
+	struct amp_mgr *mgr;
 
 	BT_DBG("%s handle 0x%2.2x status 0x%2.2x", hdev->name, ev->phy_handle,
 	       ev->status);
@@ -3426,6 +3427,14 @@ static void hci_phy_link_complete_evt(struct hci_dev *hdev,
 		return;
 	}
 
+	BT_DBG("hcon %p mgr %p", hcon, hcon->amp_mgr);
+
+	mgr = hcon->amp_mgr;
+	if (!(mgr && mgr->l2cap_conn && mgr->l2cap_conn->hcon)) {
+		hci_dev_unlock(hdev);
+		BT_DBG("Amp Manager is not Initialized");
+		return;
+	}
 	bredr_hcon = hcon->amp_mgr->l2cap_conn->hcon;
 
 	hcon->state = BT_CONNECTED;
-- 
1.7.9.5

